zaza/openstack/utilities/openstack.py

```diff
@@ -2444,6 +2444,45 @@ def upload_image_to_glance(glance, local_path, image_name, disk_format='qcow2',
     return image
 
 
+def import_image_to_glance(glance, local_path, image_name,
+                           disk_format='qcow2', visibility='public',
+                           container_format='bare'):
+    """Creates, stages and import an image.
+
+    :param glance: Authenticated glanceclient
+    :type glance: glanceclient.Client
+    :param local_path: Path to local image
+    :type local_path: str
+    :param image_name: The label to give the image in glance
+    :type image_name: str
+    :param disk_format: The of the underlying disk image.
+    :type disk_format: str
+    :param visibility: Who can access image
+    :type visibility: str (public, private, shared or community)
+    :param container_format: Whether the virtual machine image is in a file
+                             format that also contains metadata about the
+                             actual virtual machine.
+    :type container_format: str
+    :returns: glance image pointer
+    :rtype: glanceclient.common.utils.RequestIdProxy
+    """
+    image = glance.images.create(
+        name=image_name,
+        disk_format=disk_format,
+        visibility=visibility,
+        container_format=container_format)
+    glance.images.stage(image.id, open(local_path, 'rb'))
+    glance.images.image_import(image.id, method='glance-direct')
+
+    resource_reaches_status(
```

Comment:
As discussed offline, this probably doesn't wait long enough

Reply:
Images are getting cached so no need to extend waiting time ;-)